src/Core/Grid/Filter/CategoryFilterFormFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Filter;
+
+use PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinitionInterface;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\Form;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Class CategoryFilterFormFactory decorates original filter factory to add custom submit action
+ */
+final class CategoryFilterFormFactory implements GridFilterFormFactoryInterface
```

Comment:
@mickaelandrieu remember i asked about configuring filters form? for this page i needed to submit filters to custom action, so this is a workaround i came up.  what do you think about this "hack"?

Reply:
we could inject Request instead, if that would be better. :) i dont think that coupling request with form factory is that bad in this case.